ethereum/spec/src/testFixtures/java/tech/pegasys/teku/spec/util/DataStructureUtil.java

```diff
@@ -846,6 +846,16 @@ public AnchorPoint randomAnchorPoint(final UInt64 epoch) {
     return AnchorPoint.fromInitialBlockAndState(anchorBlockAndState);
   }
 
+  public AnchorPoint randomAnchorPoint(final UInt64 epoch, final Fork currentFork) {
+    UInt64 slot = computeStartSlotAtEpoch(epoch);
+    final BeaconBlockAndState blockAndState =
+        randomBlockAndState(
+            slot,
+            stateBuilderPhase0().slot(slot).build(),
+            currentFork.getBackingNode().hashTreeRoot());
```

Comment:
This is setting the parent root of the block to the current fork's hash tree root, whereas you need to put the current fork into the state:

```suggestion
        randomBlockAndState(
            slot, stateBuilderPhase0().slot(slot).fork(currentFork).build(), randomBytes32());
```